core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkerBase.scala

```diff
@@ -49,6 +56,7 @@ abstract class WorkerBase(identifier: ActorVirtualIdentity) extends WorkflowActo
   lazy val dataOutputPort: DataOutputPort = wire[DataOutputPort]
   lazy val batchProducer: TupleToBatchConverter = wire[TupleToBatchConverter]
   lazy val tupleProducer: BatchToTupleConverter = wire[BatchToTupleConverter]
+  lazy val promiseManager: WorkerPromiseManager = wire[WorkerPromiseManager]
```

Comment:
isn't needed as the promiseManager belongs in ControlInputPort

Reply:
Oh I see that this is a different type of promiseManager i.e. a WorkerPromiseManager. I still think promisemanager should be a part of inputPort and we should be wiring that here.

Reply:
promise manager is already a data member of control input port and we are wiring that here.

Reply:
Then I would ask that this be wired in WorkflowActor. I have left comments about this there too and we can discuss this